tabbycat/adjallocation/preformed/anticipated.py

```diff
@@ -26,13 +27,34 @@ def calculate_anticipated_draw(round):
 
     nteamsindebate = 4 if round.tournament.pref('teams_in_debate') == 'bp' else 2
 
-    if round.prev is None or not round.prev.debate_set.exists():
-        # Special case: If this is the first round, everyone will be on zero.
+    if round.prev is None or not round.prev.debate_set.exists() or round.is_break_round:
+        # Special cases: If this is the first round, everyone will be on zero.
         # Just take all teams, rounded down -- if this is done, it'll typically
         # be done before availability is locked down. Also do this if the last
         # round hasn't yet been drawn, since that's premature for bracket
         # predictions.
-        npanels = round.tournament.team_set.count() // nteamsindebate
+        #
+        # Also occurs for elimination rounds as everyone is just as live.
+
+        nteams = 0
+        if round.is_break_round:
+            break_size = round.break_category.break_size
+            nprev_rounds = round.break_category.round_set.filter(seq__lt=round.seq).count()
+            partial_two = nteamsindebate == 2 and not ispow2(break_size)
+            partial_bp = nteamsindebate == 4 and ispow2(break_size // 6)
+            if nprev_rounds == 0 and nteamsindebate == 2:
+                return partial_break_round_split(break_size)[0]
+            elif nprev_rounds > 0 and (partial_two or partial_bp):
+                # If using partial elimination rounds, the second round is the first for
+                # the powers of two, so start counting from here.
+                nprev_rounds -= 1
+
+            # Subsequent rounds are half the previous, but always a power of 2
+            nteams = 1 << (break_size.bit_length() - 1 - nprev_rounds)
```

Comment:
Would this be easier to read if you just treated (1) two-team break rounds, (2) BP break rounds and (3) non-existent previous rounds as wholly distinct cases, rather than this sort of nested logic?

Reply:
Fell into "smart" coding a bit here, but this shows how everything is related and similar. Treating as distinct cases wouldn't help with the number of `if`s (could even have more) as everything would be tested, but would be longer with duplication, so errors could slip in.